addons/point_of_sale/models/pos_config.py

```diff
@@ -585,10 +585,28 @@ def setup_defaults(self, company):
         :param self pos.config: pos.config records present in the company during the installation of localisation.
         :param company res.company: the single company where the pos.config defaults will be setup.
         """
-        self.assign_payment_journals(company)
-        self.generate_pos_journal(company)
-        self.setup_invoice_journal(company)
-
+        if not self.env["pos.config"].search([('company_id', '=', company.id)]):
+            self.create_pos_config_for_company(company)
+        else:
+            self.assign_payment_journals(company)
+            self.generate_pos_journal(company)
+            self.setup_invoice_journal(company)
+
+    def create_pos_config_for_company(self, company):
+        journal = self.env['account.journal'].search([('type', '=', 'purchase'), ('company_id', '=', company.id)], limit=1)
```

Comment:
@loukwets The onboarding data of classic PoS create a Point of Sale journal of type Miscellaneous, maybe it would also be good to create one. Here you link a journal that has no link to what you do in the PoS.